src/object.c

```diff
@@ -166,13 +166,12 @@ int git_object_lookup_prefix(
 			error = git_odb_read(&odb_obj, odb, id);
 		}
 	} else {
-		git_oid short_oid;
+		git_oid short_oid = {{ 0 }};
 
 		/* We copy the first len*4 bits from id and fill the remaining with 0s */
```

Comment:
This comment isn't quite accurate anymore. Maybe "We only copy over the first len*4 bits from the id"